### PR TITLE
RSA8f4

### PR DESCRIPTION
**Pull request by ricardopereira**

### CODE REVIEW

File: Spec/Auth.swift [hunk 1]
```diff
@@ -713,6 +713,33 @@ class Auth : QuickSpec {
                 }
 
             }
+
+            // RSA8f4
+            pending("ensure the message published with a wildcard '*' has the provided clientId") {
+                let token = getTestToken()
+                let options = ARTClientOptions(token: token)
+                options.environment = "sandbox"
+                options.clientId = "john"
```
Comment (mattheworiordan) [inline Spec/Auth.swift]: Why are you setting the `clientId`?

File: Spec/Auth.swift [hunk 2]
```diff
@@ -713,6 +713,33 @@ class Auth : QuickSpec {
                 }
 
             }
+
+            // RSA8f4
+            pending("ensure the message published with a wildcard '*' has the provided clientId") {
+                let token = getTestToken()
+                let options = ARTClientOptions(token: token)
+                options.environment = "sandbox"
```
Comment (mattheworiordan) [inline Spec/Auth.swift]: Is this necessary, can this not be a default in test?

File: Spec/Auth.swift [hunk 3]
```diff
@@ -713,6 +713,33 @@ class Auth : QuickSpec {
                 }
 
             }
+
+            // RSA8f4
+            pending("ensure the message published with a wildcard '*' has the provided clientId") {
+                let token = getTestToken()
+                let options = ARTClientOptions(token: token)
```
Comment (mattheworiordan) [inline Spec/Auth.swift]: The token is supposed to have a wildcard clientId yet that is not configured?

File: Spec/Auth.swift [hunk 4]
```diff
@@ -713,6 +713,33 @@ class Auth : QuickSpec {
...
+                let channel = rest.channels.get("test")
+
+                waitUntil(timeout: testTimeout) { done in
+                    channel.publish([ARTMessage(name: nil, data: "no client", clientId: "john")]) { error in
+                        expect(error).to(beNil())
+                        switch extractBodyAsJSON(mockExecutor.requests.first) {
+                        case .Failure(let error):
+                            fail(error)
+                        case .Success(let httpBody):
+                            expect(httpBody.unbox["clientId"] as? String).to(equal("john"))
```
Comment (mattheworiordan) [inline Spec/Auth.swift]: I think you should just check that the message has the correct clientId when retrieved from history

Comment (ricardopereira) [inline Spec/Auth.swift]: Done 8f1da6a.

**Comment by tcard** [comment]
LGTM

**Pull request merged**